backend/db.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+
+	_ "github.com/lib/pq"
+)
+
+// UserDB is an interface for managing user data.
+type UserDB interface {
+	CreateUser(*User) error
+	DeleteUser(int) error
+	UpdateUser(*User) error
+	GetUsers() ([]*User, error)
+	GetUserByID(int) (*User, error)
+}
+
+// PostgresDB is a concrete implementation of the UserDB interface using PostgreSQL.
+type PostgresDB struct {
+	db *sql.DB
+}
+
+// NewPostgresDB creates a new PostgresDB instance and initializes the database connection.
+func NewPostgresDB() (*PostgresDB, error) {
+	connStr := "user=postgres dbname=postgres password=shift sslmode=disable"
+	db, err := sql.Open("postgres", connStr)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := db.Ping(); err != nil {
+		return nil, err
+	}
+
+	return &PostgresDB{
+		db: db,
+	}, nil
+}
+
+// Init initializes the database by creating the user table.
+func (s *PostgresDB) Init() error {
+	return s.CreateUserTable()
+}
+
+// CreateUserTable creates the "users" table if it does not exist.
+func (s *PostgresDB) CreateUserTable() error {
+	query := `create table if not exists users (
+		id serial primary key,
+		first_name varchar(55),
+		last_name varchar(55),
+		preferred_name varchar(20),
+		email varchar(55),
+		state varchar(20),
+		image_url varchar(255),
+		role varchar(20),
+		created timestamp
+	)`
+
+	_, err := s.db.Exec(query)
+	return err
+}
+
+// CreateUser inserts a new user record into the "users" table.
+func (s *PostgresDB) CreateUser(u *User) error {
+	query := `insert into users
+		(first_name, last_name, preferred_name, email, state, image_url, role, created)
+		values ($1, $2, $3, $4, $5, $6, $7, $8)`
+
+	resp, err := s.db.Query(
```

Comment:
sqlx has a much easier and less error prone way to approach this with struct binding.